src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeReferenceResolutionUtils.java

```diff
@@ -60,101 +52,68 @@
    * exceptions that occurred in any of the supplied {@code productTypes}.
    *
    * @param productTypes the product types with expanded references.
+   * @param referenceIdToKeyCache the instance that manages cache.
    * @return a {@link List} of {@link ProductTypeDraft} built from the supplied {@link List} of
    *     {@link ProductType}.
    */
   @Nonnull
   public static List<ProductTypeDraft> mapToProductTypeDrafts(
-      @Nonnull final List<ProductType> productTypes) {
-
-    final Set<Throwable> errors = new HashSet<>();
-
-    final List<ProductTypeDraft> referenceReplacedDrafts =
-        productTypes.stream()
-            .filter(Objects::nonNull)
-            .map(
-                productType -> {
-                  final List<AttributeDefinitionDraft> referenceReplacedAttributeDefinitions;
-                  try {
-                    referenceReplacedAttributeDefinitions =
-                        replaceAttributeDefinitionsReferenceIdsWithKeys(productType);
-                  } catch (InvalidProductTypeException invalidProductTypeException) {
-                    errors.add(invalidProductTypeException);
-                    return null;
-                  }
-
-                  return ProductTypeDraftBuilder.of(productType)
-                      .attributes(referenceReplacedAttributeDefinitions)
-                      .build();
-                })
-            .filter(Objects::nonNull)
-            .collect(toList());
-
-    if (!errors.isEmpty()) {
-      throw new ReferenceReplacementException(
-          "Some errors occurred during reference replacement.", errors);
-    }
-
-    return referenceReplacedDrafts;
+      @Nonnull final List<ProductType> productTypes,
+      @Nonnull final ReferenceIdToKeyCache referenceIdToKeyCache) {
+
+    return productTypes.stream()
+        .filter(Objects::nonNull)
+        .map(
+            productType -> {
+              final List<AttributeDefinitionDraft> referenceReplacedAttributeDefinitions;
+              referenceReplacedAttributeDefinitions =
```

Comment:
inline this

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/704/commits/383c81aacae37e0c9f5b4a2ce60669aa07ea51d7